src/app/loginComponents/accounts/accounts.component.ts

```diff
@@ -1,12 +1,53 @@
 import { Component, OnInit } from '@angular/core';
-import { Subject } from 'rxjs';
+import { Location } from '@angular/common';
+import { MatTabChangeEvent } from '@angular/material';
+import { FormControl } from '@angular/forms';
+import { Router } from '@angular/router';
 
 @Component({
   selector: 'app-accounts',
   templateUrl: './accounts.component.html'
 })
 export class AccountsComponent implements OnInit {
+  public currentTab = 'accounts'; // default to the 'accounts' tab
+  selected = new FormControl(0);
+  validTabs = ['accounts', 'profiles', 'dockstore account controls', 'requests'];
+  constructor(private location: Location, private router: Router) {}
+
   ngOnInit() {
     localStorage.setItem('page', '/accounts');
+    this.parseURL(this.router.url);
```

Comment:
I probably should've mentioned this earlier, but ideally, you would use something like activatedroute queryparams

Reply:
Ah, much better; was unaware of it.